legate/core/shape.py

```diff
@@ -14,38 +14,48 @@
 #
 from __future__ import annotations
 
-from collections.abc import Iterable
 from functools import reduce
+from typing import TYPE_CHECKING, Iterable, Iterator, Optional, Union
 
+from typing_extensions import TypeAlias, overload
 
-def _cast_tuple(value, ndim):
+if TYPE_CHECKING:
+    from . import IndexSpace
+    from .runtime import Runtime
+
+Extentable: TypeAlias = Union["Shape", int, Iterable[int]]
```

Comment:
Happy to know of a better name for this

Reply:
`ExtentListLike`?

Reply:
is `Shape` a "list like" ? Maybe just `ExtentLike`?

Reply:
https://github.com/nv-legate/legate.core/pull/181/commits/a1fb957ba4c02513af6915c7ff712f44270d2bad